src/include/miopen/fusion/utils.hpp

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2022 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#pragma once
+
+#include <miopen/fusion/solvers.hpp>
+
+namespace miopen {
+namespace solver {
+namespace fusion {
+
+inline int GetOpIdx(const std::vector<std::shared_ptr<FusionOpDescriptor>>& op_map,
+                    miopenFusionOp_t op)
+{
+    int idx = 0;
+    for(const auto& ptr : op_map)
+    {
+        if(ptr->kind() == op)
+            return idx;
+        ++idx;
+    }
+    return -1;
```

Comment:
```suggestion
    auto it = std::find_if(op_map.cbegin(), op_map.cend(), [op](auto&& item){ return item->kind() == op; });
    return it == op_map.cend() ? -1 : std::distance(op_map.cbegin(), it);
```
Or if you crave for cpu ticks, you should use a regular for loop instead, like `for (auto idx = 0; idx < op_map.size(); ++idx)`. Maybe even caching op_map.size(). But std::find_if should not differ by a measurable amount inside the library.

Reply:
Fixed.